src/overlays/actors/ovl_Eff_Stk/z_eff_stk.c

```diff
@@ -28,16 +28,81 @@ ActorInit Eff_Stk_InitVars = {
     (ActorFunc)EffStk_Draw,
 };
 
-#endif
+void func_80BF0DE0(EffStk* this, PlayState* play);
 
-extern UNK_TYPE D_06009F60;
+void EffStk_Init(Actor* thisx, PlayState* play) {
+    EffStk* this = (EffStk*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Eff_Stk/EffStk_Init.s")
+    Actor_SetScale(&this->actor, 0.2f);
+    this->actionFunc = func_80BF0DE0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Eff_Stk/EffStk_Destroy.s")
+void EffStk_Destroy(Actor* thisx, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Eff_Stk/func_80BF0DE0.s")
+void func_80BF0DE0(EffStk* this, PlayState* play) {
+    s16 temp_v0;
+    u16 temp_v1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Eff_Stk/EffStk_Update.s")
+    if (Cutscene_IsCueInChannel(play, CS_CMD_ACTOR_CUE_512)) {
+        switch (play->csCtx.actorCues[Cutscene_GetCueChannel(play, 0x200U)]->id) {
+            case 2:
+                this->actor.draw = EffStk_Draw;
+                if (this->unk146 < 0x3C00) {
+                    this->unk146 = this->unk146 + 0x400;
+                    this->unk148 = Math_SinS(this->unk146) * -630.0f;
+                }
+                this->unk144++;
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Eff_Stk/EffStk_Draw.s")
+            case 3:
+                this->actor.draw = EffStk_Draw;
+                if (this->unk146 < 0x3C00) {
+                    this->unk146 = this->unk146 + 0x400;
+                    this->unk148 = Math_SinS(this->unk146) * -630.0f;
+                }
+                this->unk144--;
+                break;
+
+            default:
+                this->actor.draw = NULL;
+                this->unk146 = 0;
+                this->unk148 = 0.0f;
+                break;
+        }
+    }
+}
+
+void EffStk_Update(Actor* thisx, PlayState* play) {
+    EffStk* this = (EffStk*)thisx;
+
+    this->actionFunc(this, play);
+}
+
+void EffStk_Draw(Actor* thisx, PlayState* play) {
+    EffStk* this = (EffStk*)thisx;
+
+    s32 pad;
```

Comment:
```suggestion
    EffStk* this = (EffStk*)thisx;
    s32 pad;
```
We try to avoid spaces between variable declarations